Backend/core/comms.py

```diff
@@ -61,57 +70,102 @@ def listen_tcp(self, server_addr: str, port: int):
+                            self.latest_tstamp = unpacked_data['tstamp_unix']
+                        db.insert_data(unpacked_data)
+                        solar_car_connection['lte'] = True
+                    # If we have not received anything for more than 5 seconds restart the loop
+                    if time.time() - latest_tstamp/1000 > 5:
+                        solar_car_connection['lte'] = False
+                        break
+            except Exception:
+                print(traceback.format_exc())
+                print('Issue connecting to server')
+                continue
 
-
-    def parse_packets(self, new_data: bytes):
-        """Parse and check the length of each packet"""
-        self.__tmp_data += new_data
+    def parse_packets(self, new_data: bytes, tmp_source: str):
+        """
+        Parse and check the length of each packet
+        :param new_data: Newly received bytes from the comm channel
+        :param tmp_source: Name of tmp data source, put comm channel name here e.g. tcp, lte
+        """
+        self.__tmp_data[tmp_source] += new_data
         packets = []
-
         while True:
             # Search for the next complete data packet
             try:
-                start_index = self.__tmp_data.index(b'<bl>')
-                end_index = self.__tmp_data.index(b'</bl>')
+                start_index = self.__tmp_data[tmp_source].index(b'<bsr>')
+                end_index = self.__tmp_data[tmp_source].index(b'</bsr>')
             except ValueError:
                 break
 
             #print("start index:", start_index, "end index:", end_index)
             #print(solar_car_connection)
 
             # Extract a complete data packet
-            packets.append(self.__tmp_data[start_index + 4:end_index])
+            packets.append(self.__tmp_data[tmp_source][start_index + 5:end_index])
             # Update the remaining data to exclude the processed packet
-            self.__tmp_data = self.__tmp_data[end_index + 5:]
+            self.__tmp_data[tmp_source] = self.__tmp_data[tmp_source][end_index + 6:]
 
         # If the remaining data is longer than the expected packet length,
         # there might be an incomplete packet, so log a warning.
-        if len(self.__tmp_data) >= byte_length:
+        if len(self.__tmp_data[tmp_source]) >= byte_length:
             print("ERROR: Incomplete or malformed packet ------------------------------------")
-            self.__tmp_data = b''
+            self.__tmp_data[tmp_source] = b''
 
         return packets
 
 
 def start_comms():
     gen_format_str(config.DATAFORMAT_PATH)
-    tcp = TCP()
-    tcp.listen_tcp(config.LOCAL_IP if len(sys.argv) > 1 and sys.argv[1]=='dev' else config.CAR_IP, config.DATA_PORT)
\ No newline at end of file
+    telemetry = Telemetry()
+    # Start two live comm channels
+    vps_thread = threading.Thread(target=lambda : telemetry.remote_db_fetch("http://150.136.104.125:3000"))
```

Comment:
Should this IP address be moved out into a config variable?

Reply:
Good catch